### PR TITLE
AZP: improve CI

### PR DESCRIPTION
**Pull request by avildema**
parallel execute iodemo test

### CODE REVIEW

File: buildlib/pr/io_demo/io-demo.yml [hunk 1]
```diff
@@ -122,59 +39,78 @@ jobs:
...
-            test_time: ${{ test.Value.duration }}
-            test_intefrafe: ${{ test.Value.interface }}
-            test_ucx_tls: ${{ test.Value.tls }}
-            test_extra_run_args: ${{ test.Value.extra_run_args }}
-            initial_delay: ${{ coalesce(test.Value.initial_delay, parameters.initial_delay) }}
-            test_interference: ${{ test.Value.interference }}
-      maxParallel: 1
+  - template: jobs-io-demo.yaml
+    parameters:
+      demands: "ucx_iodemo -equals yes"
```
Comment (yosefe) [inline buildlib/pr/io_demo/io-demo.yml:44]: maybe call this tag "ucx_iodemo_exclusive"

File: buildlib/pr/io_demo/io-demo.yml [hunk 2]
```diff
@@ -122,59 +39,78 @@ jobs:
...
 
-    variables:
-      workspace: drop_$(Build.BuildId)
-      io_demo_exe: drop_$(Build.BuildId)/install/bin/io_demo
-      cycles: ${{ parameters.cycles }}
-      downtime: ${{ parameters.downtime }}
-      uptime: ${{ parameters.uptime }}
+  - template: jobs-io-demo.yaml
+    parameters:
+      demands: "ucx_iodemo_without_interference -equals yes"
```
Comment (yosefe) [inline buildlib/pr/io_demo/io-demo.yml:82]: maybe call this tag "ucx_iodemo_shared"

File: buildlib/pr/io_demo/io-demo.yml (hunk 1 above)
Comment (avildema) [reply buildlib/pr/io_demo/io-demo.yml:44]: disagree,  what's "exclusive" - no additional information ( but for CI need rename all labels in code and on hosts)

File: buildlib/pr/io_demo/io-demo.yml (hunk 2 above)
Comment (avildema) [reply buildlib/pr/io_demo/io-demo.yml:82]: too, shared - no additional information , but ucx_iodemo_without_interference say that there is label for iodemo without interference

File: buildlib/pr/io_demo/io-demo.yml (hunk 1 above)
Comment (yosefe) [reply buildlib/pr/io_demo/io-demo.yml:44]: ok

File: buildlib/pr/io_demo/io-demo.yml (hunk 2 above)
Comment (yosefe) [reply buildlib/pr/io_demo/io-demo.yml:82]: ok

**Comment by yosefe** [comment]
LGTM, pls fix merge conflicts

File: buildlib/pr/io_demo/io-demo.yml [hunk 3]
```diff
@@ -1,97 +1,3 @@
...
-        tls: "rc_x"
-        test_name: am_cx6_rc
-        extra_run_args: ""
-      "tag match on CX6/DC":
-        duration: 600
-        interface: $(roce_iface_cx6)
-        tls: "dc_x"
-        test_name: tag_cx6_dc
-        extra_run_args: ""
-      "tag match on CX6/DC with auto ECE":
```
Comment (yosefe) [inline buildlib/pr/io_demo/io-demo.yml:48]: this test is missing in new code

File: buildlib/pr/io_demo/io-demo.yml [hunk 4]
```diff
@@ -133,61 +39,77 @@ jobs:
...
-        condition: succeeded()
+  - template: jobs-io-demo.yaml
+    parameters:
+      demands: "ucx_iodemo_without_interference -equals yes"
+      name: "without_interference"
+      tests:
+        "tag match on CX4 with user memh":
+          args: "-z"
+          interference: 'No'
+          duration: 120
```
Comment (yosefe) [inline buildlib/pr/io_demo/io-demo.yml:88]: can we set "duration" and "interference" parametes once for all "without_interference" tests?
same for "with_interference"

File: buildlib/pr/io_demo/io-demo.yml [hunk 5]
```diff
@@ -133,61 +39,77 @@ jobs:
...
-          ${{ test.Key }}:
-            test_name: ${{ test.Value.test_name }}
-            test_args: ${{ test.Value.args }}
-            test_time: ${{ test.Value.duration }}
-            test_intefrafe: ${{ test.Value.interface }}
-            test_ucx_tls: ${{ test.Value.tls }}
-            test_extra_run_args: ${{ test.Value.extra_run_args }}
-            initial_delay: ${{ coalesce(test.Value.initial_delay, parameters.initial_delay) }}
-            test_interference: ${{ test.Value.interference }}
-            test_with_ece: ${{ coalesce(test.Value.ece, parameters.ece) }}
```
Comment (yosefe) [inline buildlib/pr/io_demo/io-demo.yml:157]: this is missing after change

File: buildlib/pr/io_demo/io-demo.yml [hunk 6]
```diff
@@ -1,97 +1,3 @@
...
-  - name: cycles
-    type: number
-    default: 100
-  - name: downtime
-    type: number
-    default: 5
-  - name: uptime
-    type: number
-    default: 180
-  - name: ece
```
Comment (yosefe) [inline buildlib/pr/io_demo/io-demo.yml:17]: missing

File: buildlib/pr/io_demo/io-demo.yml (hunk 5 above)
Comment (avildema) [reply buildlib/pr/io_demo/io-demo.yml:157]: No, it is ok, move to jobs-io-demo.yaml

**Review by yosefe — DISMISSED** [review]

File: buildlib/pr/io_demo/io-demo.yml (hunk 4 above)
Comment (avildema) [reply buildlib/pr/io_demo/io-demo.yml]: do via parameter , looks good

**Comment by yosefe** [comment]
@avildema can you pls squash?

**Review by yosefe — APPROVED** [review]

**Pull request merged**